File: app/src/main/java/com/taskgoapp/taskgo/data/firestore/models/BankAccount.kt
package com.taskgoapp.taskgo.data.firestore.models

import java.util.Date

/**
 * Conta bancária do vendedor para receber pagamentos
 */
data class BankAccount(
    val id: String = "",
    val userId: String = "", // ID do vendedor
    val bankName: String = "", // Nome do banco
    val bankCode: String = "", // Código do banco (ex: 001 para Banco do Brasil)
    val agency: String = "", // Agência
    val account: String = "", // Conta
    val accountType: String = "", // "CHECKING" ou "SAVINGS"
    val accountHolderName: String = "", // Nome do titular
    val accountHolderDocument: String = "", // CPF ou CNPJ do titular
    val accountHolderDocumentType: String = "", // "CPF" ou "CNPJ"
    val stripeAccountId: String? = null, // ID da conta Stripe Connect (se configurada)
    val isDefault: Boolean = false, // Conta padrão para recebimentos
    val isVerified: Boolean = false, // Conta verificada pelo Stripe
    val createdAt: Date? = null,
    val updatedAt: Date? = null
)

File: backend/src/models/BankAccount.ts
import { UUID } from 'crypto';

export interface BankAccount {
  id: UUID;
  user_id: UUID; // Apenas para role='partner'
  account_holder_name: string;
  bank_code: string; // 001=BB, 341=Itau, etc.
  bank_name: string;
  account_type: 'checking' | 'savings' | 'payment';
  agency?: string;
  account_number: string;
  account_digit?: string;
  cpf_cnpj: string;
  pix_key?: string;
  pix_key_type?: 'cpf' | 'cnpj' | 'email' | 'phone' | 'random';
  is_primary: boolean;
  verified: boolean;
  created_at: Date;
  updated_at: Date;
}
